### DISCUSSION TITLE
appStore auto-subscription listener how to get $receipt in listener handle?

### DISCUSSION THREAD

**Discussion by qadeesz** [imdhemy/laravel-in-app-purchases#382] [Q&A] [answered]
I have use **DidRenew** event for auto-subscription of AppStore and put the code in listener is here: 
so my question how to get **$receipt?** because data is comming from Event.

`/**
     * Handle the event.
     */
    public function handle(object $event): void
    {

        info($event);
        $receipt = '';

        // $providerRep = $event->getSubscription()->getProviderRepresentation();
        // json_encode($providerRep->getReceipt()) // this is for googleStore

        // Verifiy the receipt on App Store servers.
        // For non-renewable subscriptions, you can use the `verifyReceipt` method.
        // Subscription::appStore()->receiptData($receipt)->verifyReceipt();
        $receiptResponse = Subscription::appStore()->receiptData($receipt)->verify();

        // Get the receipt status
        $receiptStatus = $receiptResponse->getStatus();

        if($receiptStatus->isValid()) {
            $latestReceiptInfo = $receiptResponse->getLatestReceiptInfo();
            // You can loop all of them or either get the first one (recently purchased).
            $receiptInfo = $latestReceiptInfo[0];

            $productId = $receiptInfo->getProductId();
            $transactionId = $receiptInfo->getTransactionId();
            $originalTransactionId = $receiptInfo->getOriginalTransactionId();
            $expiresDate = $receiptInfo->getExpiresDate();
            // And so on...
        } else {
            // The receipt is invalid
        }`

**Comment by imdhemy** [comment, marked as answer]
Hi @qadeesz 

If you are using Apple Notifications v2, you don't need to validate the receipt, this is done out of the box. You can find information on [accessing the event data in the documentation](https://imdhemy.com/laravel-iap-docs/docs/get-started/event-listeners#accessing-the-event-data). You can get the benefit of the  `getProviderRepresentation()` method.
